src/sync/channel.rs

```diff
@@ -0,0 +1,1134 @@
+#![warn(missing_docs, missing_debug_implementations, rust_2018_idioms)]
```

Comment:
```suggestion
```
Seems we already apply these lints at lib.rs.
https://github.com/async-rs/async-std/blob/944e43d4bf1e228b9e9243c79a3a600d97855308/src/lib.rs#L45